### DISCUSSION TITLE
Shouldn't BinaryFormatter be replaced?

### DISCUSSION THREAD

**Discussion by Slowacki** [quartznet/quartznet#1627] [Q&A]
The application I'm working on, which uses Quartz, is failing after migrating from .NET Core 3.1 to .NET 6 due to the usage of BinaryFormatter in Quartz. I dirty-fixed it by adding 
`<EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>`
However, BinaryFormatter is a vulnerability and shouldn't be used in general. Are there any plans to replace it?

**Comment by lahma** [comment]
Quartz DB persistence also supports JSON. BinaryFormatter is required when using remoting with full framework while there is no HTTP API ready yet.
